prov/efa/src/rxr/rxr_cq.c

```diff
@@ -268,18 +268,18 @@ static inline void rxr_cq_queue_pkt(struct rxr_ep *ep,
 						     rand() %
 						     RXR_RAND_MAX_TIMEOUT);
 
-		peer->rnr_timeout_exp = 1;
 		FI_DBG(&rxr_prov, FI_LOG_EP_DATA,
 		       "initializing backoff timeout for peer: %" PRIu64
 		       " timeout: %d rnr_queued_pkts: %d\n",
 		       pkt_entry->addr, peer->timeout_interval,
 		       peer->rnr_queued_pkt_cnt);
 	} else {
-		peer->rnr_timeout_exp++;
+		peer->timeout_interval = MIN(rxr_env.max_timeout,
```

Comment:
This is only safe if rxr_env.max_timeout is less than 1/2 of MAX_INT, which is a condition we don't enforce anywhere.

Reply:
I can make the type of `rxr_env.max_timeout` uint32_t, so it is always positive.

Reply:
what is (MAX_UINT32 / 2 + 2) * 2?  Just put an upper bound for max_timeout that is well above sanity but well below 1/2 MAX_INT and call it a day.

Reply:
Ah, I see your point now. So you are worried timeout was set to `rxr_env.max_timeout` then overflow when multiplied with 2.

Sure, will add a check in initialization.

Reply:
Added the check in new revision.